### PR TITLE
DWT-351 Chemical or Biological Component Name

### PR DESCRIPTION
**Pull request by abeyonalaja**
[DWT-351](https://eaflood.atlassian.net/browse/DWT-351)

As a User, 
I need to specify the chemical or biological component name if hazardous properties are present, 
So that the specific hazardous component is identified for proper classification.

[DWT-351]: https://eaflood.atlassian.net/browse/DWT-351?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/schemas/waste.test.js [hunk 1]
```diff
@@ -693,4 +718,346 @@ describe('Receipt Schema Validation', () => {
...
+          }
+        ])
+        expect(result.error).toBeDefined()
+        expect(result.error.message).toContain(
+          'Chemical or Biological components cannot be provided when no hazardous properties are indicated'
+        )
+      })
+
+      it('should reject any arbitrary text as component name when no hazardous properties', () => {
+        const result = validateComponentName(false, [
```
[the remark this answers is not in the document]
Comment (abeyonalaja) [reply src/schemas/waste.test.js:874]: What constitutes an "invalid" name for a component?

File: src/schemas/waste.test.js [hunk 2]
```diff
@@ -89,11 +89,14 @@ describe('Receipt Schema Validation', () => {
       return receiveMovementRequestSchema.validate(payload)
     }
 
-    it('should accept valid hazardous indicator (true)', () => {
```
[the remark this answers is not in the document]
Comment (abeyonalaja) [reply src/schemas/waste.test.js:93]: simplified and removed edge case tests

File: src/schemas/waste.test.js [hunk 3]
```diff
@@ -90,130 +90,143 @@ describe('Receipt Schema Validation', () => {
...
+        '$description',
+        ({ input, expectError, errorMessage }) => {
+          const result = validateHazardous(input)
+
+          if (expectError) {
+            expect(result.error).toBeDefined()
+            expect(result.error.message).toContain(errorMessage)
+          } else {
+            expect(result.error).toBeUndefined()
           }
```
Comment (MichaelY-eSynergy) [inline src/schemas/waste.test.js:168]: Tests shouldn't contain if statements (or any conditional logic) ideally...each condition should be a separate test rather than a test case.  This is why I don't like test cases, the muddy the water a bit which what should be a test and what shouldn't.

File: src/schemas/waste.js [hunk 4]
```diff
@@ -47,62 +47,78 @@ const hazardousSchema = Joi.object({
...
-          if (value < 0) {
-            return helpers.error('number.min')
-          }
-          return value
-        }
-
-        // Check if it's a valid string
-        if (typeof value === 'string') {
-          if (value === 'Not Supplied') {
+  components: Joi.array()
```
Comment (MichaelY-eSynergy) [inline src/schemas/waste.js:50]: Can we have a broader chat as a team to decide whether putting all of the validation logic into waste.js is a good idea?

**Review by MichaelY-eSynergy — CHANGES_REQUESTED** [review]

**Review by MichaelY-eSynergy — APPROVED** [review]
